.github/workflows/python-app.yml

```
@@ -12,8 +12,28 @@ on:
     types: [published]
 
 jobs:
+  version:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v3
+        with:
+          ref: 'main'
+          ssh-key: ${{ secrets.SSH_KEY }}
+      - name: Set version
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Run script
+        run: python scripts/update_version.py ${{ env.RELEASE_VERSION }}
+      - name: Auto-commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Apply version bump
```

```suggestion
          commit_message: Apply version bump
          create_branch: true
          branch: release-${{ env.RELEASE_VERSION }}
          tagging_message: 'v${{ env.RELEASE_VERSION }}'
```
I think this further automates other parts of the releasing process, please review.

---

mh the tag would conflict with the one in `main` which is also needed to trigger the CI. now i'm unsure why that was in the releasing process in the first place

---

Great suggestions for further automation. If we remove the `tagging_message`, this will execute and have the SPDX changes in the created branch. They won't be on `main`, though. I'm thinking maybe it makes sense to do this and then auto-merge the changes to `main`

---

That or splitting the steps on the CI, keep this one for the bump and have a second one for the branch. No strong opinions/

---

On it!